cli/azd/pkg/apphost/generate.go

```diff
@@ -540,11 +583,19 @@ func (b *infraGenerator) LoadManifest(m *Manifest) error {
 			b.connectionStrings[name] = *comp.ConnectionString
 		}
 
+		var deploymentParams map[string]any
+		var deploymentSource string
+		if comp.Deployment != nil {
+			deploymentParams = comp.Deployment.Params
+			deploymentSource = filepath.Base(*comp.Deployment.Path)
+		}
 		switch comp.Type {
-		case "project.v0":
-			b.addProject(name, *comp.Path, comp.Env, comp.Bindings, comp.Args)
+		case "project.v0", "project.v1":
```

Comment:
Should we split these cases so that `project.v0` explicitly passes the `'nil` and `""` for deploymentParams and deploymentSource?